amip_interp/amip_interp.F90

```diff
@@ -146,16 +146,17 @@ module amip_interp_mod
 !-----------------------------------------------------------------------
 !----------------- Public interfaces -----------------------------------
 
-public amip_interp_init, get_amip_sst, get_amip_ice, amip_interp_new, &
-       amip_interp_del, amip_interp_type, assignment(=)
+public amip_interp_init, amip_interp_init_r4, amip_interp_init_r8, get_amip_sst, &
+     & get_amip_ice, amip_interp_new, amip_interp_del, amip_interp_type, assignment(=)
 
 !-----------------------------------------------------------------------
 !----------------- Public Data -----------------------------------
 integer :: i_sst = 1200
 integer :: j_sst = 600
-real, parameter:: big_number = 1.E30
+real(r8_kind), parameter:: big_number = 1.E30_r8_kind
 logical :: forecast_mode = .false.
-real, allocatable, dimension(:,:) ::  sst_ncep, sst_anom
+real(r8_kind), allocatable, dimension(:,:) ::  sst_ncep,    sst_anom    !< r8 versions of sst_ncep and sst_anom
+real(r4_kind), allocatable, dimension(:,:) ::  sst_ncep_r4, sst_anom_r4 !< r4 versions of sst_ncep and sst_anom
```

Comment:
I think it might be best to go with real(r8_kind) sst_ncep, sst_anom and cast it to r4_kind if necessary in the procedures.  Using sst_ncep_r4 and sst_anom_r4 would require way too many changes to the components that is out of scope with this project for now.

Reply:
`sst_ncep_r4` and `sst_anom_r4` removed in cf84442e4163db85472c07529130402009c6915c.